src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -706,6 +713,21 @@ protected override BoundNode VisitReturnStatementNoAdjust(BoundReturnStatement n
             return result;
         }
 
+        private static bool IsNullable(TypeSymbolWithAnnotations typeOpt)
+        {
+            return typeOpt?.IsNullable == true;
+        }
+
+        private static bool IsNonNullable(TypeSymbolWithAnnotations typeOpt)
+        {
+            return (object)typeOpt != null && typeOpt.IsReferenceType && typeOpt.IsNullable == false;
```

Comment:
Nit: could compress a bit like you did above: `typeOpt?.IsReferenceType == true && typeOpt.IsNullable == false`